pkgs/development/compilers/llvm/5/llvm.nix

```diff
@@ -78,9 +78,9 @@ in stdenv.mkDerivation (rec {
     "-DLLVM_ENABLE_FFI=ON"
     "-DLLVM_ENABLE_RTTI=ON"
 
-    "-DLLVM_HOST_TRIPLE=${stdenv.hostPlatform.config}"
-    "-DLLVM_DEFAULT_TARGET_TRIPLE=${stdenv.targetPlatform.config}"
-    "-DTARGET_TRIPLE=${stdenv.targetPlatform.config}"
+    "-DLLVM_HOST_TRIPLE=${stdenv.buildPlatform.config}"
```

Comment:
This is very awkward but what I had to do to get equal hashes for these commmands:

```
$ nix-instantiate -A buildPackages.buildPackages.androidndk --arg crossSystem '{config = "mips64el-apple-windows-gnu"; libc = "glibc"; }'
$ nix-instantiate -A buildPackages.androidndk --arg crossSystem '{config = "mips64el-apple-windows-gnu"; libc = "glibc"; }'
```

Reply:
llvm is being pulled in through jdk & mesa-noglu.

Reply:
Wait no they should all be host. `llvm` uses the terminology correctly here. We'd just make the last one host not target for sake of the "ensure unaffected" test.

Reply:
How do you do that? I definitely don't want ensure unaffected to break LLVM cross.

Reply:
- `LLVM_DEFAULT_TARGET_TRIPLE` is just the default. It would be the host for `ensureUnaffected`'s sake.
 - [`LLVM_TARGETS_TO_BUILD`](https://llvm.org/docs/CMake.html#llvm-specific-variables) we keep unset to build them all
 - `TARGET_TRIPLE` I don't even know whether it exists, or why we used it. @dtzWill can we remove it? `